compiler/rustc_middle/src/ty/abstract_const.rs

```diff
@@ -128,67 +30,26 @@ TrivialTypeTraversalAndLiftImpls! {
 }
 
 impl<'tcx> TyCtxt<'tcx> {
+    /// Returns a const with substs applied by
+    pub fn bound_abstract_const(
+        self,
+        uv: ty::Unevaluated<'tcx, ()>,
```

Comment:
should take `ty::WithOptConstParam` instead of the full `ty::Unevaluated`

Reply:
Should it just take the `DefId`, or is there a specific reason for `ty::WithOptConstParam`?

Reply:
ah I see why, but I guess why not just take the defid then?